### PR TITLE
Write unit tests

### PR DESCRIPTION
**Pull request by MaliyOna**

### CODE REVIEW

File: CatNote.Tests/Services/GenericServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+using CatNote.DAL.Interfaces;
+using CatNote.Tests.Services.DataForTests;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace CatNote.Tests.Services;
+
+public class GenericServiceTests
+{
```
Comment (kalachck) [inline CatNote.Tests/Services/GenericServiceTests.cs:14]: Why don't you cover other entities?

**Review by kalachck — CHANGES_REQUESTED** [review]

File: CatNote.Tests/Services/GenericServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    public GenericServiceTests()
+    {
+        _mockGenericRepository = new Mock<IGenericRepository<UserEntity>>();
+        _mockMapper = new Mock<IMapper>();
+        _userService = new GenericService<UserModel, UserEntity>(_mockMapper.Object, _mockGenericRepository.Object);
+    }
+
+    [Fact]
+    public async Task Create_CreateNewUser_CallCreateMethodOfRepositoryAndReturnUserModel()
```
Comment (kalachck) [inline CatNote.Tests/Services/GenericServiceTests.cs:27]: Read about GivenWhanThen

**Review by kalachck — CHANGES_REQUESTED** [review]

File: CatNote.Tests/Services/GenericServiceTests.cs (hunk 2 above)
Comment (blackberts) [reply CatNote.Tests/Services/GenericServiceTests.cs:27]: @kalachck 
Given-When-Then (GWT) is a semi-structured way to write down test cases

Comment (blackberts) [reply CatNote.Tests/Services/GenericServiceTests.cs:27]: image

Comment (kalachck) [inline CatNote.Tests/Services/GenericServiceTests.cs:27]: Please read about Given-When-Then

**Review by kalachck — CHANGES_REQUESTED** [review]

File: CatNote.Tests/Services/GenericServiceTests.cs (hunk 2 above)
Comment (SasukeVapes) [reply CatNote.Tests/Services/GenericServiceTests.cs:27]: Also read about FIRST principles in testing

File: CatNote.Tests/Services/AchievementGenericServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+    public AchievementGenericServiceTests()
+    {
+        _mockGenericRepository = new Mock<IGenericRepository<AchievementEntity>>();
+        _mockMapper = new Mock<IMapper>();
+        _achievementService = new GenericService<AchievementModel, AchievementEntity>(_mockMapper.Object, _mockGenericRepository.Object);
+    }
+
+    [Fact]
+    public async Task Create_CorrectModelPass_CallCreateMethodOfRepositoryAndReturnAchievementModel()
```
Comment (kalachck) [inline CatNote.Tests/Services/AchievementGenericServiceTests.cs:27]: Long test names. Simplify according to Given-When-Then.
For ex: Create_WhenModelIsCorrect_ReturnsAchievementModel

**Review by kalachck — CHANGES_REQUESTED** [review]

**Review by kalachck — APPROVED** [review]

**Pull request merged**